src/pages/commits/utils.ts

```diff
@@ -72,29 +71,25 @@ const generateBuildVariantOptionsForTaskIconsFromState = (
   state: commitsPageReducerState
 ): MainlineCommitsQueryVariables["buildVariantOptions"] => {
   const { filterState } = state;
-  const { hasTasks, hasFilters } = getFilterStatus(filterState);
-  const failingStatuses = arrayIntersection(
-    FAILED_STATUSES,
-    filterState.statuses
-  );
+  const { hasTasks, hasFilters, hasStatuses } = getFilterStatus(filterState);
 
-  // no filters should show failing icons only
-  // yes filters should show failing icons only if there are failing statuses
-  // yes task filters and status filters should show all icons
-  // yes task filters and no status filters should show all icons
+  let statusesToShow = FAILED_STATUSES as string[];
 
-  const shouldShowFailingIcons =
-    !hasFilters || (failingStatuses.length > 0 && !hasTasks);
-  const shouldShowAllIcons = hasTasks;
-  const shouldShowIcons = shouldShowFailingIcons || shouldShowAllIcons;
-
-  const statusesToShowIconsFor =
-    hasTasks && hasFilters ? filterState.statuses : failingStatuses;
+  if (hasFilters && !hasTasks && !hasStatuses) {
+    statusesToShow = arraySetDifference(statusesToShow, filterState.statuses);
+  }
```

Comment:
would `filterState.statuses` ever be anything but `[]`? because this will run only when `!hasStatuses`, so I'm not sure if `arraySetDifference` adds any functionality here